tests/operators/test_datetime.py

```diff
@@ -35,24 +34,23 @@
 INTERVAL = datetime.timedelta(hours=12)
 
 
-class TestBranchDateTimeOperator(unittest.TestCase):
+class TestBranchDateTimeOperator:
     @classmethod
-    def setUpClass(cls):
-        super().setUpClass()
+    def setup_class(cls):
 
         with create_session() as session:
             session.query(DagRun).delete()
             session.query(TI).delete()
 
-        cls.targets = [
-            (datetime.datetime(2020, 7, 7, 10, 0, 0), datetime.datetime(2020, 7, 7, 11, 0, 0)),
-            (datetime.time(10, 0, 0), datetime.time(11, 0, 0)),
-            (datetime.datetime(2020, 7, 7, 10, 0, 0), datetime.time(11, 0, 0)),
-            (datetime.time(10, 0, 0), datetime.datetime(2020, 7, 7, 11, 0, 0)),
-            (datetime.time(11, 0, 0), datetime.time(10, 0, 0)),
-        ]
+    targets = [
+        (datetime.datetime(2020, 7, 7, 10, 0, 0), datetime.datetime(2020, 7, 7, 11, 0, 0)),
+        (datetime.time(10, 0, 0), datetime.time(11, 0, 0)),
+        (datetime.datetime(2020, 7, 7, 10, 0, 0), datetime.time(11, 0, 0)),
+        (datetime.time(10, 0, 0), datetime.datetime(2020, 7, 7, 11, 0, 0)),
+        (datetime.time(10, 0, 0), datetime.time(11, 0, 0)),
```

Comment:
There is two same parameters: `(datetime.time(10, 0, 0), datetime.time(11, 0, 0))`
That a good question what we actually expected when we pass two `datetime.time` arguments and `target_lower >  target_upper` it could refers to the fact that BranchOperator work wrong with this case.

Currently this test pass due to the fact that subtests do not reset DagRuns and other stuff and use only first one, it is not a case for `@pytest.mark.parametrize` and I guess initially this test failed

Reply:
Yes when target_lower>target_upper, then some error was coming with the same issue that target_lower>target_upper.

I don't know why with subTests this error was not coming.

So i thought it was some error as lower time should be smaller.

Did i missed something?

Earlier one was correct? or i did wrong on making target_lower greater?

@Taragolis

Reply:
@Taragolis 
so that was to check testcase when some wrong dates are passed?

I did wrong by changing target_lower?